### DISCUSSION TITLE
Using react-native libraries to make a react-native library

### DISCUSSION THREAD

**Discussion by mrzslr** [callstack/react-native-builder-bob#465] [Q&A] [answered]
This might sound a bit unusual but still, I'm not sure, So this is why I'm asking here.

Let's suppose you want to make a library that uses other react-native libraries that use native codes(they are not just js files). for example, use [react-native-community/geolocation](https://github.com/michalchudziak/react-native-geolocation) to create another library that uses this library with a little bit own customized. and we call this library **"rn-geo-fancy-location"**.
And now we want to use this "rn-geo-fancy-location" inside a project.

the main question would be how the package json files would look like in the "rn-geo-fancy-location" library and the project. 
Also, if the custom "rn-geo-fancy-location"(which uses RN community geolocation lib) is inside the project, pod install won't install the pod and you can't see it in the Pods list.

Have you had any experience like this? please give your comments.

**Comment by satya164** [comment, marked as answer]
https://github.com/callstack/react-native-builder-bob#how-do-i-add-a-react-native-library-containing-native-code-as-a-dependency-in-my-library

**Reply by mrzslr** [reply]
I've tried this way. but it doesn't work. Although the dependency was added to the peerDependencies to my custom RN library. when I'm doing `pod install` in the project, the react-native-geolocation is not being added to the main project.
![image](https://github.com/callstack/react-native-builder-bob/assets/5070181/ed6f631a-c65e-48ea-a8ef-6c7a97627ae6)

**Reply by satya164** [reply]
> Although the dependency was added to the peerDependencies to my custom RN library.

It needs to be installed under `dependencies` in the app as mentioned in the linked doc.

**Reply by mrzslr** [reply]
Thank you @satya164 🙌
So, I think I was doing it correctly on the first try but was expecting to be added automatically to the main project. would be better I guess to somehow let the user/consumer know by throwing an error while installing and showing them a proper message until they install the needed dependency.

**Reply by atlj** [reply]
@mrzslr, If user didn't install a package that you have as a `peerDependency`, they get a warning during `npm install` or `yarn install`. You can check: https://nodejs.org/en/blog/npm/peer-dependencies#the-solution-peer-dependencies
